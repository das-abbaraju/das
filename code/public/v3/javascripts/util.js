/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});
