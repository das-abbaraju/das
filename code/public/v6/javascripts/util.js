/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});


util.factory('picsnotify', function($rootScope, $timeout) {
    $rootScope.warning = null;
    $rootScope.error   = null;

    return {
        showWarning: function(msg) {
            if (msg) {
                $rootScope.warning = msg;
                $timeout(function() { $rootScope.warning = null }, 3000);
            }
        },
        showError: function(msg) {
            if (msg) {
                $rootScope.error = msg;
                $timeout(function() { $rootScope.error = null }, 3000);
            }
        }
    }
});

util.factory('picsappHttp', function($http, picsnotify) {
    function handleError(errorData) {
        picsnotify.showError(errorData.error);
    }

    return {
        get: function(url, successCallback) {
            $http.get(url).success(successCallback).error(handleError);
        },
        post: function(url, data, successCallback) {
            $http.post(url, data).success(successCallback).error(handleError);
        }
    }
});
