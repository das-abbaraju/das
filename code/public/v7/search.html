<div>
    <div class="row">
        <div class="col-md-6">
            <form action="" id="search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" size="50"
                       placeholder="Enter the first part of a last name"
                       ng-change="checkSearch()"
                       ng-model="searchPrefix"/>
            </form>
        </div>
    </div>

    <div id="search-results" class="row">
        <div class="col-md-12">
            <div ng-hide="people">No search results.</div>

            <div ng-show="people">{{ people.length }} matching people</div>

            <table class="table" ng-show="people" id="table-template">
                <thead>
                <tr>
                    <th>
                        <a href="" ng-click="toggleSortOrder('last')">Name</a>
                        <span ng-show="!reverse && (predicate == 'last')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'last')">&#x25bc;</span>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('ssn')">SSN</a>
                        <span ng-show="!reverse && (predicate == 'ssn')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'ssn')">&#x25bc;</span>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('birthDate')">Birth Date</a>
                  <span ng-show="!reverse && (predicate == 'birthDate')">&#x25b2;</sp  an>
                  <span ng-show="reverse && (predicate == 'birthDate')">&#x25bc;</spa  n>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('gender')">Gender</a>
                        <span ng-show="!reverse && (predicate == 'gender')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'gender')">&#x25bc;</span>
                    </th>
                </tr>
                </thead>

                <tbody>
                <tr id="template-row"
                    ng-repeat="person in people | orderBy:predicate:reverse">
                    <td class="name">
                        <a ng-href="#/show/{{person.id}}">{{person.name}}</a>
                    </td>
                    <td class="SSN">{{person.ssn}}</td>

                    <td class="DOB">{{person.birthDate | date:'d MMM, yyyy'}}</td>
                    <td class="gender">{{person.gender | gender}}</td>
                </tr>
                </tbody>
            </table>
        </div>
    </div>
</div> <!-- search -->
