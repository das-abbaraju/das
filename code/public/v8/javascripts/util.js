/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});


util.directive('messageBox', function() {
    return {
        restrict: 'E',
        scope: {
            ngModel: '=',
            type: '@',
            icon: '@'
        },
        transclude: true,
        templateUrl: 'templates/message-box.html',

        link: function (scope, element, attrs, $timeout) {
            scope.close = function () {
                scope.ngModel = null;
            }

        }
    }// The meat of the directive goes below.
});

util.factory('picsnotify', function($rootScope, $timeout) {
    $rootScope.warning = null;
    $rootScope.error   = null;

    return {
        showWarning: function(msg) {
            if (msg) {
                $rootScope.warning = msg;
                $timeout(function() { $rootScope.warning = null }, 3000);
            }
        },
        showError: function(msg) {
            if (msg) {
                $rootScope.error = msg;
                $timeout(function() { $rootScope.error = null }, 3000);
            }
        }
    }
});

util.factory('picsappHttp', function($http, picsnotify) {
    function handleError(errorData) {
        picsnotify.showError(errorData.error);
    }

    return {
        get: function(url, successCallback) {
            $http.get(url).success(successCallback).error(handleError);
        },
        post: function(url, data, successCallback) {
            $http.post(url, data).success(successCallback).error(handleError);
        }
    }
});
