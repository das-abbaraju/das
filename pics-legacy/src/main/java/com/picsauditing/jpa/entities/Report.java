package com.picsauditing.jpa.entities;

import com.picsauditing.report.fields.FieldType;
import com.picsauditing.report.fields.ReportField;
import com.picsauditing.report.models.ModelType;
import com.picsauditing.report.tables.FieldImportance;
import org.apache.commons.collections.CollectionUtils;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import javax.persistence.Column;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

@SuppressWarnings("serial")
@Entity
@Table(name = "report")
@SQLDelete(sql = "UPDATE report SET deleted = 1 WHERE id = ?")
@Where(clause = "deleted = 0")
public class Report extends BaseTable {

	private ModelType modelType;
	private User owner;
	private String name;
	private String description;
	private String filterExpression;
	private String sql;
	private boolean deleted;
	private boolean isPublic;
    private boolean groupBy;

	private List<com.picsauditing.jpa.entities.Column> columns = new ArrayList<com.picsauditing.jpa.entities.Column>();
	private List<Filter> filters = new ArrayList<Filter>();
	private List<Sort> sorts = new ArrayList<Sort>();

	private List<ReportPermissionUser> reportPermissionUsers = new ArrayList<ReportPermissionUser>();
    private List<ReportPermissionAccount> reportPermissionAccounts = new ArrayList<ReportPermissionAccount>();
	private List<ReportUser> reportUsers = new ArrayList<ReportUser>();

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "ownerID", nullable = false)
	public User getOwner() {
		return owner;
	}

	public void setOwner(User owner) {
		this.owner = owner;
	}

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	@ReportField(importance = FieldImportance.Required, width = 200)
	public ModelType getModelType() {
		return modelType;
	}

	public void setModelType(ModelType type) {
		this.modelType = type;
	}

	@Column(nullable = false)
	@ReportField(importance = FieldImportance.Required, width = 200)
	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

    @ReportField
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	@Transient
	public String getSql() {
		return sql;
	}

	public void setSql(String sql) {
		this.sql = sql;
	}

	public boolean isDeleted() {
		return deleted;
	}

	public void setDeleted(boolean deleted) {
		this.deleted = deleted;
	}

    @ReportField
    public String getFilterExpression() {
        return filterExpression;
    }

    public void setFilterExpression(String filterExpression) {
        this.filterExpression = filterExpression;
    }

    @OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<com.picsauditing.jpa.entities.Column> getColumns() {
		return columns;
	}

	public void setColumns(List<com.picsauditing.jpa.entities.Column> columns) {
		this.columns = columns;
	}

	public void addColumn(com.picsauditing.jpa.entities.Column column) {
		column.setReport(this);
		columns.add(column);
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<Filter> getFilters() {
		return filters;
	}

	public void setFilters(List<Filter> filters) {
		this.filters = filters;
	}

	public void addFilter(Filter filter) {
		filter.setReport(this);
		filters.add(filter);
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<Sort> getSorts() {
		return sorts;
	}

	public void setSorts(List<Sort> sorts) {
		this.sorts = sorts;
	}

	public void addSort(Sort sort) {
		sort.setReport(this);
		sorts.add(sort);
	}

	@Override
	public String toString() {
		return name;
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
	public List<ReportUser> getReportUsers() {
		return reportUsers;
	}

	public void setReportUsers(List<ReportUser> reportUsers) {
		this.reportUsers = reportUsers;
	}

	public boolean hasNoColumns() {
		return CollectionUtils.isEmpty(columns);
	}

	public boolean hasNoModelType() {
		return modelType == null;
	}

	public boolean hasNoOwner() {
		return owner == null || owner.getId() == 0;
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
	public List<ReportPermissionUser> getReportPermissionUsers() {
		return reportPermissionUsers;
	}

	public void setReportPermissionUsers(List<ReportPermissionUser> reportPermissionUsers) {
		this.reportPermissionUsers = reportPermissionUsers;
	}

    @OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
    public List<ReportPermissionAccount> getReportPermissionAccounts() {
        return reportPermissionAccounts;
    }

    public void setReportPermissionAccounts(List<ReportPermissionAccount> reportPermissionAccounts) {
        this.reportPermissionAccounts = reportPermissionAccounts;
    }

    public void sortColumns() {
		Collections.sort(columns, new Comparator<com.picsauditing.jpa.entities.Column>() {
			@Override
			public int compare(com.picsauditing.jpa.entities.Column c1, com.picsauditing.jpa.entities.Column c2) {
				if (c1.getSortIndex() == c2.getSortIndex()) {
					return c1.getId() - c2.getId();
				}

				return c1.getSortIndex() - c2.getSortIndex();
			}
		});
	}

    @ReportField(type = FieldType.Boolean)
    public boolean isPublic() {
		return isPublic;
	}

	public void setPublic(boolean isPublic) {
		this.isPublic = isPublic;
	}

    @Transient
    public boolean hasGroupBy() {
        return groupBy;
    }

    public void setGroupBy(boolean groupBy) {
        this.groupBy = groupBy;
    }
}