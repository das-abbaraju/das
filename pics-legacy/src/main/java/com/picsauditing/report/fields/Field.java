package com.picsauditing.report.fields;

import java.util.HashSet;
import java.util.Set;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import javax.persistence.Transient;

import com.picsauditing.access.OpPerms;
import com.picsauditing.access.Permissions;
import com.picsauditing.dr.domain.fields.DisplayType;
import com.picsauditing.dr.domain.fields.FilterType;
import com.picsauditing.jpa.entities.Column;
import com.picsauditing.report.tables.FieldImportance;
import com.picsauditing.util.Strings;

public class Field {

	private static final Pattern FIELD_VARIABLE_PATTERN = Pattern.compile("\\{(\\w+)\\}");

	private FieldType type = FieldType.String;
	private String name;
	private String categoryTranslation;
	private String text;
	private String url;
	private int width = Column.DEFAULT_WIDTH;
	private String help;

    private String requiredJoin;
	private Class<?> fieldClass;
	private String databaseColumnName;
	private boolean visible = true;
	private boolean filterable = true;
	private boolean sortable = true;
	private String preTranslation;
	private String postTranslation;
	private String separator;
	private String drillDownField;
	private OpPerms requiredPermission = OpPerms.None;
	private FieldImportance importance = FieldImportance.Low;

    private String prefixValue;
    private String suffixValue;

	public Field(ReportField annotation) {
		type = annotation.type();
		width = annotation.width();

		if (Strings.isEmpty(annotation.url())) {
			url = null;
		} else {
			url = annotation.url();
		}

		requiredPermission = annotation.requiredPermissions();
		visible = annotation.visible();
		filterable = annotation.filterable();
		sortable = annotation.sortable();
		importance = annotation.importance();

		preTranslation = annotation.i18nKeyPrefix();
		postTranslation = annotation.i18nKeySuffix();
		if (type.getFilterType() == FilterType.Multiselect && Strings.isEmpty(preTranslation)) {
			preTranslation = type.toString();
		}
	}

	public Field(String name) {
		this.name = name;
		this.databaseColumnName = name;
	}

	public Field(String name, String databaseColumnName, FieldType type) {
		this.name = name;
		this.databaseColumnName = databaseColumnName;
		if (type == null) {
			throw new RuntimeException("type is required when creating Fields");
		}

		this.type = type;
	}

	public Field setTranslationPrefixAndSuffix(String prefix, String suffix) {
		this.preTranslation = prefix;
		this.postTranslation = suffix;
		return this;
	}

	public String getI18nKey(String value) {
		String key = value;

		if (!Strings.isEmpty(preTranslation)) {
			key = preTranslation + "." + key;
		}

		if (!Strings.isEmpty(postTranslation)) {
			key = key + "." + postTranslation;
		}

		return key;
	}

	public Set<String> getDependentFields() {
		Set<String> dependent = new HashSet<String>();
		if (Strings.isNotEmpty(url)) {
			Matcher urlFieldMatcher = FIELD_VARIABLE_PATTERN.matcher(url);

			while (urlFieldMatcher.find()) {
				dependent.add(urlFieldMatcher.group(1));
			}
		}
        if (drillDownField != null)
            dependent.add(drillDownField);

		return dependent;
	}

	public boolean isTranslated() {
		if (Strings.isEmpty(preTranslation) && Strings.isEmpty(postTranslation)) {
			return false;
		}

		return true;
	}

	public FieldType getType() {
		return type;
	}

	public void setType(FieldType type) {
		this.type = type;
	}

	@Transient
	public DisplayType getDisplayType() {
		return type.getDisplayType();
	}

	@Transient
	public FilterType getFilterType() {
		return type.getFilterType();
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public String getDatabaseColumnName() {
		return databaseColumnName;
	}

	public void setDatabaseColumnName(String databaseColumnName) {
		this.databaseColumnName = databaseColumnName;
	}

	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public String getHelp() {
		return help;
	}

	public void setHelp(String help) {
		this.help = help;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	@Transient
	public String getCategoryTranslation() {
		return categoryTranslation;
	}

	public void setCategoryTranslation(String categoryTranslation) {
		this.categoryTranslation = categoryTranslation;
	}

	public void setPreTranslation(String preTranslation) {
		this.preTranslation = preTranslation;
	}

	public void setPostTranslation(String postTranslation) {
		this.postTranslation = postTranslation;
	}

    public String getSeparator() {
        return separator;
    }

    public void setSeparator(String separator) {
        this.separator = separator;
    }

	public Field requirePermission(OpPerms opPerm) {
		requiredPermission = opPerm;
		return this;
	}

	public boolean canUserSeeQueryField(Permissions permissions) {
		if (requiredPermission == null) {
			return true;
		}

		if (requiredPermission.isNone()) {
			return true;
		}

		return permissions.hasPermission(requiredPermission);
	}

	public void setFieldClass(Class<?> fieldClass) {
		this.fieldClass = fieldClass;
	}

	public Class<?> getFieldClass() {
		return fieldClass;
	}

	public String getPreTranslation() {
		return preTranslation;
	}

	public String getPostTranslation() {
		return postTranslation;
	}

	// TODO: This is very questionable
	public boolean isVisible() {
		return visible;
	}

	public void setVisible(boolean visible) {
		this.visible = visible;
	}

	public boolean isFilterable() {
		return filterable;
	}

	public void setFilterable(boolean filterable) {
		this.filterable = filterable;
	}

	public boolean isSortable() {
		return sortable;
	}

	public void setSortable(boolean sortable) {
		this.sortable = sortable;
	}

	public FieldImportance getImportance() {
		return importance;
	}

	public void setImportance(FieldImportance importance) {
		this.importance = importance;
	}

    public String getPrefixValue() {
        return prefixValue;
    }

    public void setPrefixValue(String prefixValue) {
        this.prefixValue = prefixValue;
    }

    public String getSuffixValue() {
        return suffixValue;
    }

    public void setSuffixValue(String suffixValue) {
        this.suffixValue = suffixValue;
    }

    public String getDrillDownField() {
        return drillDownField;
    }

    public void setDrillDownField(String drillDownField) {
        this.drillDownField = drillDownField;
    }

    public Field clone() {
		Field copiedField = new Field(name, databaseColumnName, type);
		copiedField.text = text;
		copiedField.url = url;
		copiedField.width = width;
		copiedField.help = help;
		copiedField.fieldClass = fieldClass;
		copiedField.visible = visible;
		copiedField.filterable = filterable;
		copiedField.sortable = sortable;
        copiedField.requiredJoin = requiredJoin;
		copiedField.preTranslation = preTranslation;
		copiedField.postTranslation = postTranslation;
        copiedField.separator = separator;
        copiedField.drillDownField = drillDownField;
		copiedField.requiredPermission = requiredPermission;
		copiedField.importance = importance;
        copiedField.prefixValue = prefixValue;
        copiedField.suffixValue = suffixValue;

		return copiedField;
	}

	@Override
	public String toString() {
		return categoryTranslation + ": " + name;
	}

    public String getRequiredJoin() {
        return requiredJoin;
    }

    public void setRequiredJoin(String requiredJoin) {
        this.requiredJoin = requiredJoin;
    }
}