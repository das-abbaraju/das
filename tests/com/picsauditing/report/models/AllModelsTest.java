package com.picsauditing.report.models;

import com.google.common.base.Joiner;
import com.picsauditing.EntityFactory;
import com.picsauditing.access.OpPerms;
import com.picsauditing.access.Permissions;
import com.picsauditing.jpa.entities.Report;
import com.picsauditing.report.SqlBuilder;
import com.picsauditing.report.fields.Field;
import com.picsauditing.search.SelectSQL;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@UseReporter(DiffReporter.class)
public class AllModelsTest {

    private AbstractModel model;
    private Permissions permissions;

    @Before
    public void setUp() {
        permissions = EntityFactory.makePermission();
    }

    @Test
	@Ignore
    public void testAllModels() throws Exception {
        StringBuilder actual = new StringBuilder();

        for (ModelType type : ModelType.values()) {
            actual.append(String.format("Model: %s\nPermission: %s\n----------------------------------------\n", type, permissions));
            permissions = EntityFactory.makePermission();
            model = ReportModelFactory.build(type, permissions);
            if (model != null) {
                actual.append(getJoin());
            } else {
                actual.append("Not Yet Implemented");
            }
            actual.append("\n\n");
        }
        Approvals.verify(actual.toString());
    }

    @Test
    @Ignore
    public void testAllSql() throws Exception {
        SqlBuilder builder = new SqlBuilder();
        StringBuilder actual = new StringBuilder();

        for (ModelType type : ModelType.values()) {
            actual.append(String.format("Model: %s\nPermission: %s\n----------------------------------------\n", type, permissions));
            permissions = EntityFactory.makePermission();
            try {
                final Report report = new Report();
                model = ReportModelFactory.build(type, permissions);
                report.setModelType(type);
                final SelectSQL selectSQL = builder.initializeReportAndBuildSql(report, permissions);
                actual.append(selectSQL.toString());
            } catch (Exception e) {
                actual.append("Not Yet Implemented");
            }
            actual.append("\n\n");
        }
        Approvals.verify(actual.toString());
    }

    @Test
    public void testContractorsModel_Admin() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllOperators);
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        permissions.setAccountType("Admin");
        model = new ContractorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorsModel_Billing() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.Billing);
        permissions.setAccountType("Admin");
        model = new ContractorsModel(permissions);
        String output = "Permissions: " + permissions + "\n" + getJoin();
        Approvals.verify(output);
    }

    @Test
    public void testContractorsModel_Corporate() throws Exception {
        permissions.setAccountType("Corporate");
        model = new ContractorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorAuditOperatorsModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new ContractorAuditOperatorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorFeesModel_Billing() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.Billing);
        permissions.setAccountType("Admin");
        model = new ContractorFeesModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testInvoiceModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new InvoicesModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testPaymentModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new PaymentsModel(permissions);
        Approvals.verify(getJoin());
    }

    private String getJoin() {
        List<String> fieldValues = new ArrayList<String>();

        for (Field field : model.getAvailableFields().values()) {
            fieldValues.add(field.getName());
        }
        Collections.sort(fieldValues);
        return Joiner.on("\n").join(fieldValues);
    }

}
