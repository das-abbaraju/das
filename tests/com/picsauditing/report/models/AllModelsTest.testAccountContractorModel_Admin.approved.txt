ContactInformation: AccountAddress
ContactInformation: AccountAddress2
ContactInformation: AccountAddress3
ContactInformation: AccountCity
ContactInformation: AccountContactDepartment
ContactInformation: AccountContactEmail
ContactInformation: AccountContactFax
ContactInformation: AccountContactFirstName
ContactInformation: AccountContactID
ContactInformation: AccountContactLastName
ContactInformation: AccountContactLocale
ContactInformation: AccountContactName
ContactInformation: AccountContactPhone
ContactInformation: AccountContactTimezone
ContactInformation: AccountCountry
ContactInformation: AccountCountrySubdivision
AccountInformation: AccountCreationDate
Billing: AccountDeactivationDate
ContactInformation: AccountFax
ContactInformation: AccountFullAddress
AccountInformation: AccountID
AccountInformation: AccountLegalName
Classification: AccountMaterialSupplier
Classification: AccountNaicsCode
CompanyStatistics: AccountNaicsDart
CompanyStatistics: AccountNaicsLwcr
CompanyStatistics: AccountNaicsTrir
AccountInformation: AccountName
Classification: AccountOffsiteServices
Classification: AccountOnsiteServices
ContactInformation: AccountPhone
Billing: AccountReason
Classification: AccountRequiresCompetencyReview
Classification: AccountRequiresOQ
AccountInformation: AccountStatus
General: AccountTimezone
Classification: AccountTransportationServices
ContactInformation: AccountWebUrl
ContactInformation: AccountZip
Billing: ContractorAccountLevel
Billing: ContractorAuditGUARD
Billing: ContractorBalance
CustomerService: ContractorCustomerServiceUserID
CustomerService: ContractorCustomerServiceUserName
Billing: ContractorDocuGUARD
CustomerService: ContractorDontReassign
Billing: ContractorEmployeeGUARD
CustomerService: ContractorInsideSalesUserID
CustomerService: ContractorInsideSalesUserName
Billing: ContractorInsureGUARD
AccountInformation: ContractorMembershipDate
DocumentsAndAudits: ContractorPQFID
DocumentsAndAudits: ContractorPQFTypeName
Billing: ContractorPayingFacilities
Classification: ContractorProductRisk
Classification: ContractorProductRiskVerified
CustomerService: ContractorRecommendedCSRID
CustomerService: ContractorRecommendedCSRName
Billing: ContractorRenew
Classification: ContractorSafetyRisk
Classification: ContractorSafetyRiskVerified
CompanyStatistics: ContractorScore
Classification: ContractorSoleProprietor
AccountInformation: ContractorTagTagID
Billing: ContractorTaxId
Classification: ContractorTradesSelf
Classification: ContractorTradesSub
Classification: ContractorTransportationRisk
Classification: ContractorTransportationRiskVerified
AccountInformation: ContractorWatchwatch
DocumentsAndAudits: ContractorWelcomeCallCaoID
DocumentsAndAudits: ContractorWelcomeCallCaoStatus
DocumentsAndAudits: ContractorWelcomeCallID
DocumentsAndAudits: ContractorWelcomeCallTypeName